src/components/TradeWidget/index.tsx

```diff
@@ -690,8 +717,46 @@ const TradeWidget: React.FC = () => {
     receiveValue,
   ])
 
+  const fallBackNetworkId = networkId || Network.Mainnet // fallback to mainnet
```

Comment:
we have this logic in many places, since we get the networkId always from `useWalletConnection` should we move the logic there and expose `networkIdOrDefault` for example?

Reply:
Maybe in a different PR, too many places to change